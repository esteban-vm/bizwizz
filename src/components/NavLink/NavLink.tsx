import type { FC, UUID, SectionID } from '@/types'
import Link from 'next/link'

export interface NavLinkProps {
  id: UUID
  text: Capitalize<string>
  className: `nav_link ${'header' | 'footer'}`
  link: `#${Extract<SectionID, 'contact' | 'portfolio' | 'services'>}`
}

const NavLink: FC<NavLinkProps> = ({ link, text, ...props }) => {
  return (
    <li {...props}>
      <Link href={link}>{text}</Link>
    </li>
  )
}

export default NavLink
