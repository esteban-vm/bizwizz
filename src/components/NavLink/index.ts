export { default as NavLink, type NavLinkProps } from './NavLink'
