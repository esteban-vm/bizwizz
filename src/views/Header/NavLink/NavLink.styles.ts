import clsx from 'clsx'

const styles = clsx(
  'ml-10 block select-none pb-[0.8rem] ',
  'text-[1.4rem] uppercase text-white transition-[border-b]',
  'hover:border-b-2 hover:border-b-primary'
)
export default styles
