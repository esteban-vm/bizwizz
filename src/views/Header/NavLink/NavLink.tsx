import type { FC, UUID } from '@/types'
import Link from 'next/link'
import styles from './NavLink.styles'

export type NavLinkProps = {
  id: UUID
  link: `#${Lowercase<string>}`
  text: Capitalize<string>
}

const NavLink: FC<NavLinkProps> = ({ id, link, text }) => {
  return (
    <li id={id}>
      <Link className={styles} href={link}>
        {text}
      </Link>
    </li>
  )
}

export default NavLink
