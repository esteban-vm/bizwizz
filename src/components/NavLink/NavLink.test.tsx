import { render, cleanup, within, screen } from '@/tests'
import NavLink, { type NavLinkProps } from './NavLink'

describe('Nav Link:', () => {
  let asFragment: () => DocumentFragment

  const props: NavLinkProps = {
    className: 'header_nav_link',
    id: '31c97b93-9ad9-4064-9a41-d8034c2faccf',
    link: '#contact',
    text: 'Contact',
  }

  beforeEach(() => {
    void ({ asFragment } = render(<NavLink {...props} />))
  })

  afterEach(cleanup)

  it('should render correctly', () => {
    expect(asFragment()).toMatchSnapshot()
  })

  it('should have correct structure', () => {
    const navLinkElem = screen.getByRole('listitem')
    expect(navLinkElem).toBeInTheDocument()
    expect(navLinkElem).toHaveAttribute('id', props.id)
    expect(navLinkElem).toHaveAttribute('class', props.className)
    const linkElem = within(navLinkElem).getByRole('link', { name: props.text })
    expect(linkElem).toBeInTheDocument()
    expect(linkElem).toHaveAttribute('href', props.link)
  })
})
