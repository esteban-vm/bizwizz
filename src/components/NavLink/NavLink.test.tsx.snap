// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Nav Link: > should render correctly 1`] = `
<DocumentFragment>
  <li
    class="header_nav_link"
    id="31c97b93-9ad9-4064-9a41-d8034c2faccf"
  >
    <a
      href="#contact"
    >
      Contact
    </a>
  </li>
</DocumentFragment>
`;
